### PR TITLE
[16.0] Fix multi databases with list_db = false 

### PR DESCRIPTION
**Pull request by guewen**
Port of https://github.com/OCA/queue/pull/379 , https://github.com/OCA/queue/pull/571, #580

### CODE REVIEW

**Comment by pcastelovigo** [comment]
thank you so much

**Review by rvalyi — APPROVED** [review]

File: queue_job/jobrunner/runner.py [hunk 1]
```diff
@@ -398,10 +398,16 @@ def from_environ_or_config(cls):
         return runner
 
     def get_db_names(self):
+        """
+        >>> runner = QueueJobRunner()
+        >>> config["db_name"] = None
+        >>> runner.get_db_names()
+        ['odoo']
+        """
```
Comment (sim6) [inline queue_job/jobrunner/runner.py:406]: What about add `list_db = False` to test `odoo.service.db.list_dbs(True)` against `odoo.service.db.exp_list(True)`?
```
>>> config["list_db"] = False
```

Comment (pcastelovigo) [reply queue_job/jobrunner/runner.py:406]: i'm not sure about the implications, as I must confess I never checked what list_dbs or exp_list does

Comment (pcastelovigo) [reply queue_job/jobrunner/runner.py:406]: oh, i finally understood. Yes, thats a great idea, and a much better test approach
This only checks list_dbs gives something like untested exp_lists, your test checks if it works without list_dbs

Comment (sim6) [reply queue_job/jobrunner/runner.py:406]: :) Please see https://github.com/OCA/queue/pull/591

**Pull request merged**